node/pkg/near/watcher.go

```diff
@@ -0,0 +1,369 @@
+package near
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/p2p"
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	"github.com/certusone/wormhole/node/pkg/readiness"
+	"github.com/certusone/wormhole/node/pkg/supervisor"
+	"github.com/certusone/wormhole/node/pkg/vaa"
+	eth_common "github.com/ethereum/go-ethereum/common"
+	"github.com/mr-tron/base58"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/tidwall/gjson"
+	"go.uber.org/zap"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type (
+	// Watcher is responsible for looking over Near blockchain and reporting new transactions to the wormhole contract
+	Watcher struct {
+		nearRPC          string
+		wormholeContract string
+
+		msgChan  chan *common.MessagePublication
+		setChan  chan *common.GuardianSet
```

Comment:
Remove the setChan bits please. This was just removed from the terra^Wcosmoasm watcher in #1350 since it is only authoritative on ethereum.